### PR TITLE
Account overview

### PR DESCRIPTION
**Pull request by rBangay**
A/B test of the new account overview product home and all subsequent journeys.

ticket(s):
https://trello.com/c/P43U9JJM/686-goal-demoable-account-overview-on-mma-in-code-1-3
https://trello.com/c/a6Tr6KGz/947-goal-subscription-detail-pages-for-new-account-overview-2-3
https://trello.com/c/uYNCVOAu/948-goal-release-account-overview

### Account overview
0-accountoverview-1

### Manage subscription
Screenshot_2020-05-18 My Account The Guardian

### Payment update
1-paymentupdate-1

### Payment update cont.
1-paymentupdate-2

### Payment update cont.
1-paymentupdate-3

### Payment update cont.
1-paymentupdate-4

### Delivery address update
1-addressupdate-1

### Delivery address update cont.
1-addressupdate-2

### Delivery address update cont.
1-addressupdate-3

### Delivery problem reporting
1-deliveryhistory-3

### Delivery problem reporting cont.
1-deliveryhistory-4

### Delivery problem reporting cont.
1-deliveryhistory-1

### Delivery problem reporting cont.
1-deliveryhistory-2

### Manage membership
Screenshot_2020-05-18 My Account The Guardian(1)

### Cancel membership
Screenshot_2020-05-18 My Account The Guardian

### Cancel membership cont.
Screenshot_2020-05-18 My Account The Guardian

### Manage contribution
Screenshot_2020-05-18 My Account The Guardian(1)

### Change contribution amount
Screenshot_2020-05-18 My Account The Guardian

### Change contribution amount cont.
Screenshot_2020-05-18 My Account The Guardian(1)

### Cancel contribution
Screenshot_2020-05-18 My Account The Guardian

### Cancel contribution cont.
Screenshot_2020-05-18 My Account The Guardian(1)

### Account overview - no products
image

### Manage subscription - payment failure
image

more screenshots on their way ...

### CODE REVIEW

File: app/client/components/accountoverview/accountOverview.tsx [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
+  const membershipData = productDetailList.filter(
+    _ => _.topLevelProductType === ProductTypes.membership
+  );
+
+  if (productDetailList.length === 0) {
+    return <EmptyAccountOverview />;
+  }
+
+  // TODO: check to see if babel polyfils find
```
[the remark this answers is not in the document]
Comment (rBangay) [reply app/client/components/accountoverview/accountOverview.tsx:56]: Yep we did, it is polyfilled by the babel/polyfill dependency which we're already using .. you got in there with this comment before I noticed it and removed it!

File: app/client/components/user.tsx [hunk 2]
```diff
@@ -87,7 +101,6 @@ const User = () => (
             key={productType.urlPart}
             path={"/cancel/" + productType.urlPart}
             productType={productType}
-            currentStep={1}
```
[the remark this answers is not in the document]
Comment (rBangay) [reply app/client/components/user.tsx:90]: Yep the new step indicator takes a static object as config. So yep it's hard coded on the respective components

File: app/client/components/buttons.tsx [hunk 3]
```diff
@@ -47,10 +49,10 @@ const applyIconStyleIfApplicable = (
...
   }
   return {
-    padding: "1px 15px 0 15px",
-    svg: {
-      display: "none"
-    }
+    padding: `1px ${space[5]}px 0 ${space[5]}px`
+    // svg: {
+    //   display: "none",
+    // },
```
[the remark this answers is not in the document]
Comment (rBangay) [reply app/client/components/buttons.tsx:55]: This was changed here to allow the subsequent addition of a warning/error icon inside the button ... alongside removing the display:none of all svg's. I have also changed the adding of the svgs to be conditional based on the props already passed through therefore making this css unnecessary. I should have removed this comment though :s

File: app/client/components/delivery/records/userPhoneNumber.tsx [hunk 4]
```diff
@@ -193,7 +193,6 @@ export const UserPhoneNumber = (props: UserPhoneNumberProps) => {
               <>
                 <TextInput
                   label="Phone number"
-                  supporting="Enter your phone number"
```
[the remark this answers is not in the document]
Comment (rBangay) [reply app/client/components/delivery/records/userPhoneNumber.tsx:196]: :s I don't know why this happened .. adding back in

File: app/client/components/page.tsx [hunk 5]
```diff
@@ -1,10 +1,12 @@
...
 import { breakpoints } from "@guardian/src-foundations";
 import { palette, space } from "@guardian/src-foundations";
-import React from "react";
+import { headline, textSans } from "@guardian/src-foundations/typography";
+import { Link } from "@reach/router";
+import React, { ReactElement } from "react";
 import { minWidth } from "../styles/breakpoints";
-import { titlepiece } from "../styles/fonts";
-import { gridBase, gridItemPlacement } from "../styles/grid";
+// import { titlepiece } from "../styles/fonts";
```
[the remark this answers is not in the document]
Comment (rBangay) [reply app/client/components/page.tsx:8]: I'll add the TODO in there ... this is indeed the font that needs adding back in when we get to the font usage overhaul

File: app/client/components/payment/cardDisplay.tsx [hunk 6]
```diff
@@ -50,6 +59,13 @@ export const CardDisplay = (props: CardDisplayProps) => (
...
     }}
   >
-    {cardTypeToSVG(props.type)}•••• •••• •••• {props.last4}
+    {cardTypeToSVG(props.type)}{" "}
+    <span
+      css={css`
+        color: ${props.inErrorState ? palette.news[400] : palette.neutral[7]};
+      `}
+    >
+      Ending {props.last4}
```
[the remark this answers is not in the document]
Comment (rBangay) [reply app/client/components/payment/cardDisplay.tsx:68]: This Change will affect overview and payment flow pages. 

Previously the display looked like this

`•••• •••• •••• 1234`

and now looks like this:

`Ending 1234`

File: app/client/components/delivery/records/userPhoneNumber.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (rBangay) [reply app/client/components/delivery/records/userPhoneNumber.tsx]: re added back in

File: app/client/components/holiday/holidayQuestionsModal.tsx [hunk 7]
```diff
@@ -6,7 +6,7 @@ import { Modal } from "../modal";
 import { InfoIcon } from "../svgs/infoIcon";
 
 export const creditExplainerSentence = (issueKeyword: string) =>
-  `You will be credited for each suspended ${issueKeyword} on the next bill after the ${issueKeyword} date.`;
+  `You will be credited for each suspended ${issueKeyword} on your next bill.`;
```
[the remark this answers is not in the document]
Comment (rBangay) [reply app/client/components/holiday/holidayQuestionsModal.tsx:9]: reverted

File: app/client/components/payment/cardDisplay.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (rBangay) [reply app/client/components/payment/cardDisplay.tsx:68]: Changed case to `ending`

File: app/client/components/accountoverview/manageMembership.tsx [hunk 8]
```diff
@@ -0,0 +1,269 @@
...
+    <FlowStartMultipleProductDetailHandler
+      {...props}
+      headingPrefix={"Manage"}
+      hideHeading
+      hasLeftNav={{
+        pageTitle: "Manage contribution",
+        selectedNavItem: navLinks.accountOverview
+      }}
+      supportRefererSuffix="manage_contribution_flow"
+      loadingMessagePrefix="Retrieving details of your"
```
Comment (twrichards) [inline app/client/components/accountoverview/manageMembership.tsx:49]: don't think this is correct, given we're in `manageMembership.tsx`

File: app/client/components/accountoverview/manageSubscription.tsx [hunk 9]
```diff
@@ -0,0 +1,360 @@
...
+    <FlowStartMultipleProductDetailHandler
+      {...props}
+      headingPrefix={"Manage"}
+      hideHeading
+      hasLeftNav={{
+        pageTitle: "Delivery history",
+        selectedNavItem: navLinks.accountOverview
+      }}
+      supportRefererSuffix="delivery_records_flow"
+      loadingMessagePrefix="Retrieving details of your"
```
Comment (twrichards) [inline app/client/components/accountoverview/manageSubscription.tsx:52]: I'm thinking...
image

**Pull request merged**